site/content/blog/carvelizing-helm-chart.md

```diff
@@ -0,0 +1,455 @@
+stringData:
+  values.yml: |
+    ---
+    image:
+      pullPolicy: Always
+    serverBlock: |-
+      server {
+        listen 0.0.0.0:8080;
+        location / {
+          return 200 "Response from Custom Server";
+        }
+      }
+```
+
+```bash
+$ kapp deploy -a pkg-demo -f pkgInstall.yaml -y
+Target cluster 'https://192.168.64.32:8443' (nodes: minikube)
+
+Changes
+
+Namespace  Name          Kind            Conds.  Age  Op      Op st.  Wait to    Rs  Ri
+default    nginx-pkg     PackageInstall  -       -    create  -       reconcile  -   -
+^          nginx-values  Secret          -       -    create  -       reconcile  -   -
+
+Op:      2 create, 0 delete, 0 update, 0 noop, 0 exists
+Wait to: 2 reconcile, 0 delete, 0 noop
+
+12:38:14PM: ---- applying 1 changes [0/2 done] ----
+12:38:14PM: create secret/nginx-values (v1) namespace: default
+12:38:14PM: ---- waiting on 1 changes [0/2 done] ----
+12:38:14PM: ok: reconcile secret/nginx-values (v1) namespace: default
+12:38:14PM: ---- applying 1 changes [1/2 done] ----
+12:38:14PM: create packageinstall/nginx-pkg (packaging.carvel.dev/v1alpha1) namespace: default
+12:38:14PM: ---- waiting on 1 changes [1/2 done] ----
+12:38:14PM: ongoing: reconcile packageinstall/nginx-pkg (packaging.carvel.dev/v1alpha1) namespace: default
+12:38:14PM:  ^ Waiting for generation 1 to be observed
+12:38:15PM: ongoing: reconcile packageinstall/nginx-pkg (packaging.carvel.dev/v1alpha1) namespace: default
+12:38:15PM:  ^ Reconciling
+12:39:15PM: ---- waiting on 1 changes [1/2 done] ----
+12:39:16PM: ongoing: reconcile packageinstall/nginx-pkg (packaging.carvel.dev/v1alpha1) namespace: default
+12:39:16PM:  ^ Reconciling
+12:39:31PM: ok: reconcile packageinstall/nginx-pkg (packaging.carvel.dev/v1alpha1) namespace: default
+12:39:31PM: ---- applying complete [2/2 done] ----
+12:39:31PM: ---- waiting complete [2/2 done] ----
+
+Succeeded
+```
+
+After the deploy has finished, kapp-controller will have installed the package in the cluster. We can verify this by doing the `kapp inspect` and checking the pods to see that we have a workload pod running.
+```bash
+$ kapp inspect -a pkg-demo
+Target cluster 'https://192.168.64.32:8443' (nodes: minikube)
+
+Resources in app 'pkg-demo'
+
+Namespace  Name          Kind            Owner  Conds.  Rs  Ri  Age
+default    nginx-pkg     PackageInstall  kapp   1/1 t   ok  -   7m
+^          nginx-values  Secret          kapp   -       ok  -   7m
+
+Rs: Reconcile state
+Ri: Reconcile information
+
+2 resources
+
+Succeeded
+$ kubectl get pods
+NAME                         READY   STATUS    RESTARTS   AGE
+nginx-pkg-6db5c6978d-tt8k4   1/1     Running   0          8m
+```
+
+Once the pod is ready, you can use kubectl’s port forwarding to verify the customized response used in the workload.
+
+```bash
+$ kubectl port-forward service/nginx-pkg 3000:80 &
+```
+
+Now if we make a request against our service, we can see that server response is ```Response from Custom Server```
```

Comment:
nit: Should it be "request to"?

Reply:
grammar nit: "that server response is" => "that the server response is"

Reply:
I think it is fine.